RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate.cc

```diff
@@ -133,36 +133,24 @@ bool SiPixelTemplate::pushfile(int filenum, std::vector< SiPixelTemplateStore >
    // Local variables
    int i, j, k, l;
    float qavg_avg;
-   const char *tempfile;
    //	char title[80]; remove this
    char c;
    const int code_version={17};
-   
-   
-   
    //  Create a filename for this run
-   
-   std::ostringstream tout;
-   
    //  Create different path in CMSSW than standalone
-   
+   std::string tempfile=std::to_string(filenum);
+   for (unsigned cnt=4-tempfile.length(); cnt > 0; cnt-- ){ tempfile ="0"+tempfile; }
+   tempfile = "template_summary_zp" + tempfile + ".out";
```

Comment:
@pmaksim1 , the default file name generated here should be the same if SI_PIXEL_TEMPLATE_STANDALONE is defined

Reply:
@smuzaffar 
I thought I fixed the default directory in #22458 a week ago Monday.  But I can try merging this with the current code.  (I really have no preference how the strings are concatenated.)  But, do we know why this fixes the problem with the string concatenation?

Anyway, and much more importantly, the default directory CANNOT be the same when SI_PIXEL_TEMPLATE_STANDALONE since CalibTracker/SiPixelESProducers/data/ does not exist.  (And it does not because we are not in the CMSSW release... That's what standalone means.)

So how do we fix this?  I can merge SiPixelTemplate.cc and send to @mrodozov for approval and commit?

Reply:
fwiw - the loop just above can be replaced by something simple like
tempfile=std::string(4-tempfile.length(),"0").append(tempfile)

the undying issue with the original implementation is likely using a c_str of a temporary object at line 154.
file.fullPath()).c_str() [better just to avoid char*s...]